sdk/schemaregistry/azure-data-schemaregistry-jsonschema/src/main/java/com/azure/data/schemaregistry/jsonschema/SchemaRegistryJsonSchemaSerializer.java

```diff
@@ -0,0 +1,345 @@
+import com.azure.core.http.ContentType;
+import com.azure.core.models.MessageContent;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.serializer.TypeReference;
+import com.azure.data.schemaregistry.SchemaRegistryAsyncClient;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Class that serializes and deserializes objects using <a href="https://json-schema.org/">JSON schema</a>.
+ *
+ * @see SchemaRegistryJsonSchemaSerializerBuilder
+ */
+public final class SchemaRegistryJsonSchemaSerializer {
+    private static final String CONTENT_TYPE = ContentType.APPLICATION_JSON;
+    private static final SerializerEncoding ENCODING = SerializerEncoding.JSON;
+
+    private final ClientLogger logger = new ClientLogger(SchemaRegistryJsonSchemaSerializer.class);
+    private final SchemaRegistrySchemaCache schemaCache;
+    private final JsonSchemaGenerator schemaGenerator;
+    private final SerializerAdapter serializer;
+
+    SchemaRegistryJsonSchemaSerializer(SchemaRegistryAsyncClient schemaRegistryClient,
+        JsonSchemaGenerator schemaGenerator, SerializerOptions serializerOptions) {
+        Objects.requireNonNull(schemaRegistryClient, "'schemaRegistryClient' cannot be null.");
+        Objects.requireNonNull(serializerOptions, "'serializerOptions' cannot be null.");
+
+        this.schemaGenerator = Objects.requireNonNull(schemaGenerator, "'schemaGenerator' cannot be null.");
+
+        this.serializer = serializerOptions.getSerializerAdapter();
+        this.schemaCache = new SchemaRegistrySchemaCache(schemaRegistryClient, serializerOptions.getSchemaGroup(),
+            serializerOptions.autoRegisterSchemas(), serializerOptions.getMaxCacheSize());
+    }
+
+    /**
+     * Serializes the object into a message.  Tries to infer the schema definition based on the object.  If no schema
+     * cannot be inferred, no validation is performed.
+     *
+     * @param object Object to serialize.
+     * @param typeReference Type reference of message to create.
+     * @param <T> Type of message to serialize.
+     *
+     * @return The object serialized into a message content.  If the inferred schema definition does not exist or the
+     *     object does not match the existing schema definition, an exception is thrown.
+     */
+    public <T extends MessageContent> T serialize(Object object, TypeReference<T> typeReference) {
+        return serializeAsync(object, typeReference).block();
```

Comment:
not blocking first beta, but we should add fully-sync implementation and leverage our new sync stack benefits

Reply:
Created an issue for this: https://github.com/Azure/azure-sdk-for-java/issues/36430